kbase-extension/static/kbase/js/common/jobManager.js

```diff
@@ -492,53 +454,71 @@ define([
             /**
              * @param {Object} message
              */
-            handleJobStatus(self, message) {
+            handleJobStatus(self, message, addressArgs) {
+                const updated = [];
+                Object.values(message).forEach((jobStatusMessage) => {
+                    if (!jobStatusMessage.error) {
+                        updated.push(self._handleJobStatus(jobStatusMessage, addressArgs));
+                    }
+                });
+                // update the state as appropriate
+                const toUpdate = updated.filter((job) => {
+                    return !!job;
+                });
+                if (toUpdate.length) {
+                    self.updateModel(toUpdate);
+                }
+            }
+
+            _handleJobStatus(message, addressArgs) {
                 const { jobState } = message,
                     { status } = jobState,
                     jobId = jobState.job_id;
 
                 // check if the job object has changed since we last saved it
-                const savedState = self.model.getItem(`exec.jobs.byId.${jobId}`);
+                const savedState = this.model.getItem(`exec.jobs.byId.${jobId}`);
                 if (savedState && _.isEqual(savedState, jobState)) {
+                    // return nothing if the state does not need to be updated
                     return;
                 }
 
                 // if the job is in a terminal state and cannot be retried,
                 // stop listening for updates
                 if (Jobs.isTerminalStatus(status) && !Jobs.canRetry(jobState)) {
-                    self.removeListener(jobId, jcm.MESSAGE_TYPE.STATUS);
-                    if (status === 'does_not_exist') {
-                        self.removeChannelListeners(jobId);
+                    if (addressArgs.channelType === jcm.CHANNEL.JOB) {
+                        this.removeListener(jobId, jcm.MESSAGE_TYPE.STATUS);
```

Comment:
The logic involving removing job listeners is only relevant if job messages are using channel type JOB with the job ID as the channel ID. DATAUP-497 enables sending messages by batch ID, so all jobs within a batch use the same channel, one with channel type BATCH and the channel ID set to the batch parent ID. In that case, we don't need to do so much listener cleanup as there are only five listeners (status, info, retry, logs, error) for the whole batch.